### PR TITLE
Throw ApiErrorException instead of returning false

### PR DESCRIPTION
**Pull request by WillSewell**
Note this introduces a breaking change to all of the methods
that wrap API requests.

Advantages:

- Removes the need for the `debug` parameter to return the error
  message
- Makes it more obvious when requests fail. Previously a user
  would only see an error if they had enabled a logger.
- It's more idiomatic to throw exceptions than return `false`
  in modern PHP. For example the Stripe PHP SDK does it this way:
  https://github.com/stripe/stripe-php/blob/master/lib/Exception/ApiErrorException.php

Other related changes:
- The trigger methods now return an empty object rather than `true`
  on success. This more accurately reflects what the API returns.
- The `get` method now has an `$associative` parameter that controls
  how `json_decode` decodes the response body.
- The `get` method does not include the whole response from curl,
  but just the decoded response body.
- Renamed `$response` -> `$result` in some tests. I think this is more
  consistent with the name given to the root object in the response bodies.
- removed `PusherTriggerUnitTest::testNullSocketID` - this was hard to
  test now that the subject-under-test throws exceptions due to invalid
  credentials rather than just returning null. I think this case is sufficiently
  covered in the unit tests, so I removed it.

Resolves https://github.com/pusher/pusher-http-php/issues/229

TODO

- [x] Update https://pusher.com/docs (if necessary)

### CODE REVIEW

**Comment by WillSewell** [comment]
@GrahamCampbell I'd be really interested in hearing any feedback you might have? This is a pretty substantial breaking change so I want the improvement to be worth it!

File: tests/acceptance/channelQueryTest.php [hunk 1]
```diff
@@ -69,31 +69,27 @@ public function testFilterByPrefixOneChannel()
 
     public function testUsersInfo()
     {
-        $response = $this->pusher->get_users_info('presence-channel-test');
+        $result = $this->pusher->get_users_info('presence-channel-test');
 
-        // print_r( $response );
+        // print_r( $result );
```
Comment (jonathanlloyd) [inline tests/acceptance/channelQueryTest.php:74]: Should probably just remove this!

**Review by jonathanlloyd — APPROVED** [review]
Makes sense to me!

File: tests/acceptance/channelQueryTest.php (hunk 1 above)
Comment (WillSewell) [reply tests/acceptance/channelQueryTest.php:74]: It's done all over the place, so I'll remove in a future PR

**Pull request merged**